.github/workflows/snapshot-release.yml

```diff
@@ -50,4 +52,121 @@ jobs:
             Version: `${{ steps.set-snapshot-version.outputs.VERSION }}`
             Repository: `https://s01.oss.sonatype.org/content/repositories/snapshots/`
           edit-mode: replace
-          token: ${{ secrets.GITHUB_TOKEN }}
\ No newline at end of file
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+  # This job depends on the publish job to finish so we can use the snapshot version to verify
+  # gradle compatibility of the sample apps with actual releases of the SDK with latest changes.
+  gradle-compatibility-builds:
+    name: ${{ matrix.sample-app }} build with AGP ${{ matrix.agp-version }}
+    needs: publish
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # Matrix of sample apps and AGP versions to test
+        # List all sample apps here that you want to test with the AGP versions.
+        sample-app-with-agp:
+          - "java_layout_current"
+          - "kotlin_compose_current"
+          - "java_layout_7"
+          - "kotlin_compose_7"
+        include:
+          - sample-app-with-agp: "java_layout_current"
+            sample-app: "java_layout"
+            cio-siteid-secret-key: "CUSTOMERIO_JAVA_WORKSPACE_SITE_ID"
+            cio-apikey-secret-key: "CUSTOMERIO_JAVA_WORKSPACE_API_KEY"
+            agp-version: "8.3.1"
+            # We don't need to specify more versions here, as we'll use the default values for actual gradle versions.
+          - sample-app-with-agp: "kotlin_compose_current"
+            sample-app: "kotlin_compose"
+            cio-siteid-secret-key: "CUSTOMERIO_KOTLIN_WORKSPACE_SITE_ID"
+            cio-apikey-secret-key: "CUSTOMERIO_KOTLIN_WORKSPACE_API_KEY"
+            agp-version: "8.3.1"
+            # We don't need to specify more versions here, as we'll use the default values for actual gradle versions.
+          - sample-app-with-agp: "java_layout_7"
+            sample-app: "java_layout"
+            cio-siteid-secret-key: "CUSTOMERIO_JAVA_WORKSPACE_SITE_ID"
+            cio-apikey-secret-key: "CUSTOMERIO_JAVA_WORKSPACE_API_KEY"
+            agp-version: "7.4.1"
```

Comment:
`minifyReleaseWithR8` does break with `7.2.0` so that verifies this does help us confirm if it works correctly with Gradle 7 (latest release) 🎉